### PR TITLE
Use faster Buffer.BlockCopy

### PR DESCRIPTION
**Pull request by jkotas**
See https://github.com/dotnet/corefx/pull/3615 on why this is better.

### CODE REVIEW

**Comment by jkotas** [comment]
@KrzysztofCwalina PTLA

File: src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs [hunk 1]
```diff
@@ -494,7 +494,7 @@ public void CopyTo(byte[] buffer)
             }
             else
             {
-                _bytes.CopyTo(buffer, 0);
```
Comment (stephentoub) [inline src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs]: Is there any further benefit to Buffer.BlockCopy, since these are byte arrays?

**Comment by stephentoub** [comment]
LGTM

File: src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs (hunk 1 above)
Comment (jkotas) [inline src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs]: Thanks - changed it to Buffer.BlockCopy

**Pull request merged**